packages/cashier/src/Stores/__tests__/general-store.spec.js

```diff
@@ -0,0 +1,520 @@
+    });
+
+    it('should cahange value of the variable is_cashier_default', () => {
+        general_store.setIsCashierDefault(true);
+
+        expect(general_store.is_cashier_default).toBeTrue();
+    });
+
+    it('should set deposit target', () => {
+        general_store.setDepositTarget('/cashier/payment-agent');
+
+        expect(general_store.deposit_target).toBe('/cashier/payment-agent');
+    });
+
+    it('should route to deposit target', () => {
+        general_store.setDepositTarget('/cashier/payment-agent');
+        general_store.continueRoute();
+
+        expect(general_store.root_store.common.routeTo).toHaveBeenCalledWith('/cashier/payment-agent');
+    });
+
+    it('should perform proper setAccountSwitchListener invocation', () => {
+        const spyDisposeSwitchAccount = jest.spyOn(general_store, 'disposeSwitchAccount');
+        const spyOnSwitchAccount = jest.spyOn(general_store, 'onSwitchAccount');
+        general_store.setAccountSwitchListener();
+
+        expect(spyDisposeSwitchAccount).toHaveBeenCalledTimes(1);
+        expect(spyOnSwitchAccount).toHaveBeenCalledAfter(spyDisposeSwitchAccount);
+        expect(spyOnSwitchAccount).toHaveBeenCalledWith(general_store.accountSwitcherListener);
+    });
+
+    it('should perform proper init invocation when is_logged_in is equal to true', async () => {
+        const spyGetAdvertizerError = jest.spyOn(general_store, 'getAdvertizerError');
+        const spyCheckP2pStatus = jest.spyOn(general_store, 'checkP2pStatus');
+        general_store.root_store.client.is_logged_in = true;
+        general_store.init();
+
+        await waitFor(() => {
+            expect(spyGetAdvertizerError).toHaveBeenCalledTimes(1);
+        });
+        expect(spyCheckP2pStatus).toHaveBeenCalledTimes(1);
+        expect(general_store.root_store.modules.cashier.withdraw.check10kLimit).toHaveBeenCalledTimes(1);
+    });
+
+    it('should set advetiser error', async () => {
+        const spySetP2pAdvertiserError = jest.spyOn(general_store, 'setP2pAdvertiserError');
+        await general_store.getAdvertizerError();
+
+        expect(spySetP2pAdvertiserError).toHaveBeenCalledWith('advetiser_error');
+    });
+
+    it('should set p2p advetiser error', () => {
+        general_store.setP2pAdvertiserError('p2p_advetiser_error');
+
+        expect(general_store.p2p_advertiser_error).toBe('p2p_advetiser_error');
+    });
+
+    it('should set is_p2p_visible equal to false, if there is a "RestrictedCountry" p2p advetiser error', () => {
+        general_store.setP2pAdvertiserError('RestrictedCountry');
+        general_store.checkP2pStatus();
+
+        expect(general_store.is_p2p_visible).toBeFalse();
+    });
+
+    it('should set is_p2p_visible equal to false, if there is a "RestrictedCurrency" p2p advetiser error', () => {
+        general_store.setP2pAdvertiserError('RestrictedCurrency');
+        general_store.checkP2pStatus();
+
+        expect(general_store.is_p2p_visible).toBeFalse();
+    });
```

Comment:
The test is duplicated.

Reply:
@farrah-deriv there are two different test cases: first for "RestrictedCountry" error and second for "RestrictedCurrency" error

Reply:
Oh i thought they are the same. Thanks for clarifying.